feat(web): add plugin playground layer example

ShogoHirasawa opened the pull request:
# Overview
This is PR to marge Plugin Playground example. I made "Add CZML", "Add KML", "Add CSV" and "Add 3D Tiles from URL"
## What I've done
- [x] Add CZML
- [x] Add KML
- [x] Add CSV
- [x] Add 3D Tiles from URL  

## What I haven't done
- [ ] Add OSM 3D Tiles
- [ ] Add MVT 
- [ ] Add WMS
## How I tested
I tested them in my local environment. Here are the captures for each example.
**Add CZML**
<img width="400" alt="image" src="https://github.com/user-attachments/assets/def42f77-5290-4c53-b7f7-c38958be0ac2" />   
**Add KML**
<img width="400" alt="image" src="https://github.com/user-attachments/assets/bbfbf22a-18b2-424a-9fa2-07721c49e8df" />   
**Add CSV**
<img width="400" alt="image" src="https://github.com/user-attachments/assets/79919020-2cd6-4cbd-9f52-82382299b164" />   
**Add CSV**
<img width="400" alt="image" src="https://github.com/user-attachments/assets/acb2c1a7-9fbd-43cd-9529-3024766d1bb9" />
**Add 3D Tiles**
<img width="400" alt="image" src="https://github.com/user-attachments/assets/8c5d9d9a-c8b4-446a-addc-13785007db09" />
## Which point I want you to review particularly
- Whether Each example work well
- Whether Each script have enough comment
- Whether Each example have enough datasets
## Memo


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Added support for multiple layer plugins in the Plugin Playground.
	- Introduced new layer types:
		- 3D Tiles
		- CSV
		- CZML
		- KML
	- Removed the previous layers category and restructured plugin organization.
	- Expanded layer capabilities to support various data formats and visualization options.
	- Enhanced the existing GeoJSON layer by simplifying its configuration.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

airslice commented on web/src/beta/features/PluginPlayground/Plugins/presets/layers/add-csv.ts:
```diff
@@ -0,0 +1,52 @@
+  data: {
+    type: "csv",
+    url: "https://reearth.github.io/visualizer-plugin-sample-data/public/csv/marker.csv", // URL of CSV
+    csv: { // Define by column number or column name
```
```suggestion
    csv: { // Define by column name
```
What's column number? index?

ShogoHirasawa replied:
I mean I want to say "column index ". Sorry for the confusion.

airslice replied:
No problem. Please note that we don't support define by index here.

airslice commented on web/src/beta/features/PluginPlayground/Plugins/presets/layers/add-3Dtiles.ts:
```diff
@@ -0,0 +1,74 @@
+    type: widget
+    name: Add 3D Tiles
+    description: Add 3D Tiles
+    widgetLayout:
+      defaultLocation:
+        zone: outer
+        section: left
+        area: top
```
```suggestion
```
1. This widget doesn't provide a ui, so don't define this
2. In most cases, using default location is enough

airslice commented on web/src/beta/features/PluginPlayground/Plugins/presets/layers/add-csv.ts:
```diff
@@ -0,0 +1,52 @@
+    type: widget
+    name: Add CSV
+    description: Add CSV
+    widgetLayout:
+      defaultLocation:
+        zone: outer
+        section: left
+        area: top
```
```suggestion
```
Same as above

airslice commented on web/src/beta/features/PluginPlayground/Plugins/presets/layers/add-czml.ts:
```diff
@@ -0,0 +1,83 @@
+    type: widget
+    name: Add CZML
+    description: Add CZML
+    widgetLayout:
+      defaultLocation:
+        zone: outer
+        section: left
+        area: top
```
```suggestion
```
Same as above

airslice commented on web/src/beta/features/PluginPlayground/Plugins/presets/layers/add-kml.ts:
```diff
@@ -0,0 +1,65 @@
+    type: widget
+    name: Add KML
+    description: Add KML
+    widgetLayout:
+      defaultLocation:
+        zone: outer
+        section: left
+        area: top
```
```suggestion
```
Same as above

airslice commented on web/src/beta/features/PluginPlayground/Plugins/presets/layers/add-czml.ts:
```diff
@@ -0,0 +1,83 @@
+
+// Difine encoded CZML in a variable
+const encodedCzml =
+  "data:text/plain;charset=UTF-8,%5B%0A%20%20%7B%0A%20%20%20%20%22id%22%3A%20%22document%22%2C%0A%20%20%20%20%22name%22%3A%20%222D%20Polygon%20Example%22%2C%0A%20%20%20%20%22version%22%3A%20%221.0%22%0A%20%20%7D%2C%0A%20%20%7B%0A%20%20%20%20%22id%22%3A%20%22polygon%22%2C%0A%20%20%20%20%22name%22%3A%20%22Polygon%20Example%22%2C%0A%20%20%20%20%22polygon%22%3A%20%7B%0A%20%20%20%20%20%20%22positions% ...
```
Please define the original data and using an encode funtion to get the data url. It's difficult for user to understand and modify the data if it's already encoded.

airslice commented on web/src/beta/features/PluginPlayground/Plugins/presets/layers/add-kml.ts:
```diff
@@ -0,0 +1,65 @@
+
+// Define encoded KML in a variable
+const encodedKml =
+  "data:text/plain;charset=UTF-8,%3C%3Fxml%20version%3D%221.0%22%20encoding%3D%22UTF-8%22%3F%3E%0A%3Ckml%20xmlns%3D%22http%3A%2F%2Fwww.opengis.net%2Fkml%2F2.2%22%3E%0A%20%20%3CPlacemark%3E%0A%20%20%20%20%3Cname%3EPolygon%3C%2Fname%3E%0A%20%20%20%20%3CPolygon%3E%0A%20%20%20%20%20%20%3CouterBoundaryIs%3E%0A%20%20%20%20%20%20%20%20%3CLinearRing%3E%0A%20%20%20%20%20%20%20%20%20%20%3Ccoordinates%3E%0A ...
```
Please define the original data and using an encode funtion to get the data url. It's difficult for user to understand and modify the data if it's already encoded.

airslice commented on web/src/beta/features/PluginPlayground/Plugins/presets/layers/add-kml.ts:
```diff
@@ -0,0 +1,65 @@
+  id: "layers-add-kml-reearth-yml",
+  title: "reearth.yml",
+  sourceCode: `id: layers-add-kml-plugin
+name: Add CZML
+version: 1.0.0
```
Please fix this

airslice commented on web/src/beta/features/PluginPlayground/Plugins/presets/layers/add-kml.ts:
```diff
@@ -0,0 +1,65 @@
+  data: {
+    type: "kml", 
+    url: "https://reearth.github.io/visualizer-plugin-sample-data/public/kml/square.kml", // URL of KML file
+  },
```
```suggestion
  },
  polygon: {}
```
Better keep the same structure.

ShogoHirasawa commented:
### What I fixed
- Delete "widgetLayout" setting
- Add process to define original code of czml and kml and then encode
- Added  polygon style definition of kml

mulengawilfred reviewed (commented):
No comments from me outside the issues raised by Shaun and coderrabit-AI which would need addressing.
